crates/isograph_compiler/src/source_files.rs

```diff
@@ -0,0 +1,353 @@
+                        let value = occupied_entry.into_mut();
+                        *value = document;
+                    }
+                    Entry::Vacant(vacant_entry) => {
+                        vacant_entry.insert(document);
+                    }
+                };
+            }
+            ChangedFileKind::JavaScriptSourceFile => {
+                let canonicalized_root_path = get_canonicalized_root_path(&config.project_root)?;
+                let (path_buf, file_content) =
+                    read_file(path.to_path_buf(), &canonicalized_root_path)?;
+                let (file_path, iso_literals) =
+                    read_and_parse_iso_literals(path_buf, file_content, &canonicalized_root_path)?;
+                if iso_literals.is_empty() {
+                    self.contains_iso.remove(&file_path);
+                } else {
+                    match self.contains_iso.entry(file_path) {
+                        Entry::Occupied(occupied_entry) => {
+                            let value = occupied_entry.into_mut();
+                            *value = iso_literals;
+                        }
+                        Entry::Vacant(vacant_entry) => {
+                            vacant_entry.insert(iso_literals);
+                        }
+                    };
+                }
+            }
+        }
+        Ok(())
+    }
+
+    pub fn handle_rename(
+        &mut self,
+        config: &CompilerConfig,
+        source_path: &Path,
+        target_path: &PathBuf,
+        change_kind: &ChangedFileKind,
+    ) -> Result<(), BatchCompileError> {
+        match change_kind {
+            ChangedFileKind::Schema => {
+                if config.schema != *target_path {
+                    return Err(BatchCompileError::SchemaNotFound);
+                }
+            }
+            ChangedFileKind::SchemaExtension => {
+                if config.schema_extensions.contains(target_path) {
+                    let (file_path, document) = read_and_parse_schema_extensions(target_path)?;
+                    match self.schema_extensions.entry(file_path) {
+                        Entry::Occupied(occupied_entry) => {
+                            let value = occupied_entry.into_mut();
+                            *value = document;
+                        }
+                        Entry::Vacant(vacant_entry) => {
+                            vacant_entry.insert(document);
+                        }
+                    };
+                } else {
+                    let interned_file_path = intern_file_path(source_path);
+                    self.schema_extensions.remove(&interned_file_path);
+                }
+            }
+            ChangedFileKind::JavaScriptSourceFile => {
+                let source_file_path = intern_file_path(source_path);
+                let target_file_path = intern_file_path(target_path);
+                if let Some(entry) = self.contains_iso.remove(&source_file_path) {
```

Comment:
The parsed iso contains a Location, which is to say, it contains the interned file path :/  This isn't ideal, we should fix that. But in the mean time, maybe it makes sense to treat renames as delete 1 + create 2